yt_dlp/extractor/piramidetv.py

```diff
@@ -0,0 +1,104 @@
+from .common import InfoExtractor
+from ..utils import parse_iso8601, smuggle_url, unsmuggle_url, url_or_none
+from ..utils.traversal import traverse_obj
+
+
+class PiramideTVIE(InfoExtractor):
+    _VALID_URL = r'https?://piramide\.tv/video/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://piramide.tv/video/wWtBAORdJUTh',
+        'info_dict': {
+            'id': 'wWtBAORdJUTh',
+            'ext': 'mp4',
+            'title': 'md5:79f9c8183ea6a35c836923142cf0abcc',
+            'description': '',
+            'thumbnail': 'https://cdn.jwplayer.com/v2/media/W86PgQDn/thumbnails/B9gpIxkH.jpg',
+            'channel': 'León Picarón',
+            'channel_id': 'leonpicaron',
+            'timestamp': 1696460362,
+            'upload_date': '20231004',
+        },
+    }, {
+        'url': 'https://piramide.tv/video/wcYn6li79NgN',
+        'info_dict': {
+            'id': 'wcYn6li79NgN',
+            'ext': 'mp4',
+            'title': 'ACEPTO TENER UN BEBE CON MI NOVIA\u2026? | Parte 1',
+            'description': '',
+            'channel': 'ARTA GAME',
+            'channel_id': 'arta_game',
+            'thumbnail': 'https://cdn.jwplayer.com/v2/media/cnEdGp5X/thumbnails/rHAaWfP7.jpg',
+            'timestamp': 1703434976,
+            'upload_date': '20231224',
+        },
+    }]
+
+    def _extract_video(self, video_id):
+        if video_data := self._download_json(
+                f'https://hermes.piramide.tv/video/data/{video_id}', video_id, fatal=False):
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                f'https://cdn.piramide.tv/video/{video_id}/manifest.m3u8', video_id, fatal=False)
+            next_video = traverse_obj(video_data, ('video', 'next_video', 'id', {str}))
+            return next_video, {
+                'id': video_id,
+                'formats': formats,
+                'subtitles': subtitles,
+                'webpage_url': f'https://piramide.tv/video/{video_id}',
+                'original_url': f'https://piramide.tv/video/{video_id}',
```

Comment:
Extractors should not set `original_url`.
We also don't want to overwrite `webpage_url` as the info saved in the infodict should be usable to rerun the extraction exactly as during the initial run.
```suggestion
```